sphinxcontrib/katex-server.js

```diff
@@ -0,0 +1,140 @@
+    // Split the input stream into individual rendering requests
+    // How many bytes are still left in the current request
+    let bytesLeft = null;
+    // List of chunks belonging to the current request
+    let buffer = [];
+    // Any undecoded bytes from the previous chunk, for example we were expecting
+    // a 4-byte integer but only received 2 bytes
+    let undecodedBytes = null;
+    client.on("data", function(chunk) {
+        // If there are any undecoded bytes left from the previous chunk, prepend
+        // them to the current one
+        if (undecodedBytes !== null) {
+            chunk = Buffer.concat([undecodedBytes, chunk]);
+            undecodedBytes = null;
+        }
+
+        let start = 0;
+        while (start < chunk.length) {
+            let remainingBytes = chunk.length - start;
+            if (bytesLeft === null) {
+                if (remainingBytes >= 4) {
+                    bytesLeft = chunk.readInt32LE(start);
+                    start += 4;
+                } else {
+                    undecodedBytes = chunk.slice(start);
+                    break;
+                }
+            } else {
+                if (remainingBytes >= bytesLeft) {
+                    let end = start + bytesLeft;
+                    buffer.push(chunk.slice(start, end));
+
+                    var all = Buffer.concat(buffer);
+                    handleRequest(client, all.toString());
+
+                    bytesLeft = null;
+                    buffer = [];
+                    start = end;
+                } else {
+                    buffer.push(chunk.slice(start));
+
+                    bytesLeft -= remainingBytes;
+                    break;
+                }
+            }
+        }
+    });
+
+    client.on("end", function() {
+        buffer = [];
+    });
+}
+
+function handleRequest(client, serialized) {
+    try {
+        var request = JSON.parse(serialized);
+    } catch (e) {
+        let response = { "error": `Could not deserialize ${serialized}: ${e.message}` };
+        sendMessage(client, JSON.stringify(response));
+
+        return;
+    }
+
+    try {
+        let latex = request["latex"];
+        let options = request["katex_options"] || {};
+        // this is where math latex equation is processed
+        let html = katex.renderToString(latex, options);
+
+        let response = { "html": html };
+        sendMessage(client, JSON.stringify(response));
+    } catch (e) {
+        let response = { "error": e.message };
+        sendMessage(client, JSON.stringify(response));
+    }
+}
+
+function sendMessage(client, message) {
+    // Encode the message string into bytes
+    let msgBuffer = Buffer.from(message, "utf-8");
+
+    // Tell the client how many bytes we are going to send
+    let lengthBuffer = Buffer.alloc(4);
+    lengthBuffer.writeInt32LE(msgBuffer.length, 0);
+    client.write(lengthBuffer);
+
+    // Send the actual message
+    client.write(msgBuffer);
+}
```

Comment:
Please add newline at the end of the file

Reply:
I did not know how to include JS files in `setup.py` when I created the PR. I will update it with new code. I will update the PR with your comments.

Reply:
I will add this.